File: sdk/android/component/sqljson/controller/NotificationController.java
package ardvro.component.sqljson.controller;

import com.google.gson.internal.LinkedTreeMap;

import java.util.ArrayList;

import ardvro.core.connection.IConnection;
import ardvro.core.connection.IResponseListener;
import ardvro.core.transceiver.Box;

public class NotificationController
{
    private IConnection _conection;

    public NotificationController(IConnection connection)
    {
        _conection = connection;
    }

    /*public void Send(String receiver, String message, String url, String note, String type, IResponseListener onResponse)
    {
        String functionName = "ardvro/component/sqljson/notification/Send";
        _conection.Submit(new Object[]{ receiver, message, type, url, note }, functionName, onResponse);
    }*/

    public void UpdateStatus(Object id, String status, IResponseListener<Integer> onResponse)
    {
        String functionName = "ardvro/component/sqljson/notification/UpdateStatus";
        _conection.Submit(new Object[]{ id, status }, functionName, onResponse);
    }

    public void CountUserNotification(IResponseListener<Integer> onResponse)
    {
        String functionName = "ardvro/component/sqljson/notification/CountUserNotification";
        _conection.Submit(new Object[]{ }, functionName, onResponse);
    }

    public void GetUserNotifications(int pageIndex, int pageSize, IResponseListener<Integer> onResponse)
    {
        String functionName = "ardvro/component/sqljson/notification/GetUserNotifications";
        _conection.Submit(new Object[]{ pageIndex, pageSize }, functionName, onResponse);
    }

    public void Broadcast(LinkedTreeMap<String, Object> data, IResponseListener<ArrayList<Box>> onResponse)
    {
        String functionName = "ardvro/component/sqljson/notification/Broadcast";
        _conection.Submit(data, functionName, onResponse);
    }

    public void IsOnline(String username, IResponseListener<Boolean> onResponse)
    {
        String functionName = "ardvro/component/sqljson/notification/IsOnline";
        _conection.Submit(new Object[]{ }, functionName, onResponse);
    }

}

File: sdk/web/component/sqljson/controller/NotificationController.js
var NotificationController = function NotificationController(cfg)
{
    let ctrl = {};

    /*ctrl.Send = function (receiver, message, url, note, type, onResponse)
    {
        let args = [receiver, message, type, url, note];
        cfg.Connector.Submit(args, "ardvro/component/sqljson/Notification/Send", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };*/

    ctrl.UpdateStatus = function (id, status, onResponse)
    {
        let args = [id, status];
        cfg.Connector.Submit(args, "ardvro/component/sqljson/Notification/UpdateStatus", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.CountUserNotification = function (onResponse)
    {
        cfg.Connector.Submit([], "ardvro/component/sqljson/Notification/CountUserNotification", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.GetUserNotifications = function (pageIndex, pageSize, onResponse)
    {
        cfg.Connector.Submit([pageIndex, pageSize], "ardvro/component/sqljson/Notification/GetUserNotifications", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.Broadcast = function (flex, onResponse)
    {
        cfg.Connector.Submit(flex, "ardvro/component/sqljson/Notification/Broadcast", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.IsOnline = function (username, callback)
    {
        cfg.Connector.Submit(username, "ardvro/component/sqljson/Notification/IsOnline", function (box)
        {
            callback != null ? callback(box == null ? null : box.Data) : null;
        });
    };


    return ctrl;
};